### PR TITLE
Refactory spring gateway plugin

### PR DESCRIPTION
**Pull request by ascrutae**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#4588

### CODE REVIEW

**Comment by ascrutae** [comment]
Link #4887

**Comment by wu-sheng** [comment]
A little thing could be improved
1. Webflux operation should
1. Tags of exit spans.

**Comment by ascrutae** [comment]
Link: #4668

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/tag/Tags.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ private Tags() {
     public static final StringTag MQ_TOPIC = new StringTag(9, "mq.topic");
 
     public static final class HTTP {
-        public static final StringTag METHOD = new StringTag(10, "http.method");
+        public static final StringTag METHOD = new StringTag(10, "http.method", true);
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/tag/Tags.java:81]: Please don't open this as overridable. It costs too much.

**Pull request merged**